virtual-display-driver-cli/src/main.rs

```diff
@@ -0,0 +1,343 @@
+use clap::Parser;
+use client::Client;
+use joinery::JoinableIterator;
+use lazy_format::lazy_format;
+use owo_colors::OwoColorize;
+use serde::{Deserialize, Serialize};
+
+mod client;
+
+#[derive(Debug, Parser)]
+struct Args {
+    /// Format output as JSON.
+    #[clap(short, long)]
+    json: bool,
+
+    #[clap(subcommand)]
+    command: Command,
+}
+
+#[derive(Debug, Parser)]
+enum Command {
+    /// List currently connected virtual monitors.
+    List,
+    /// Add a new virtual monitor.
+    Add {
```

Comment:
This question is mostly about UX: since the resolution is a flag, it's pretty common for flag ordering on UNIX to be unordered. The Clap derive stuff seems to bake that in, but I'm not sure about vanilla Clap parsing (I've only ever used the derive-based tools).

Another idea I had was to instead specify each mode as a single argument, like this:

```
virtual-display-driver-cli.exe add 1280x720 1920x1080@60 3840x2160@60/30
```

With that interface, I think it becomes really easy and intuitive to add any combination of modes (and would then make `add-mode` and `remove-mode` clearer too). That would still be compatible with an `-r` flag to add a "shared refresh rate" as well. The downside is it's less "script-friendly" as you'd then need to do string interpolation to add a mode from two numbers, but maybe that's a minor concern.

Reply:
_Marking as resolved, this will be handled in a follow-up_

Reply:
> Another idea I had was to instead specify each mode as a single argument, like this:

I like how that looks! It is easy to type out and clear on what it does.

I guess we can only make this so easy in the end. Maybe the easiest way for scripting is to only add shared refresh rates instead of having to deal with all of those? Admittingly it's a little less fine grained control though